### PR TITLE
Improve Prow setup documents

### LINKED ISSUE #1855: Improve the prow_setup doc
The [prow_setup](https://github.com/knative/test-infra/blob/master/guides/prow_setup.md) doc is now outdated, and some words are not accurate. We need to improve it.

/assign

### PR DESCRIPTION
**Pull request by chizhg**
**What this PR does, why we need it**:
1. Make some improvements to the Prow setup document, as some contents are outdated and are causing confusion to the developers. Fixes #1855 
2. Prow config updater also updates boskos config changes.

/cc @chaodaiG

### CODE REVIEW

File: guides/prow_setup.md [hunk 1]
```diff
@@ -1,12 +1,43 @@
...
+[prow-staging.knative.dev](https://prow-staging.knative.dev) for testing the new
+config changes before rolling out to production, and all its config files are
+under [config/prow-staging](../config/prow-staging).
 
-1. Check under [prow/cluster](../config/prow/cluster) and add a new file there
-   if not exist.
+## Modify Prow cluster configs
+
+Prow is a collection of microservices that are deployed on a Kubernetes
+cluster,
```
Comment (mattmoor) [inline guides/prow_setup.md:12]: Remove trailing whitespace:
```suggestion
cluster,
```

File: guides/prow_knative_setup.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
+# Prow setup for Knative projects
```
Comment (chaodaiG) [inline guides/prow_knative_setup.md:1]: Is this just copy/pasted from original place?

File: guides/prow_setup.md [hunk 3]
```diff
@@ -1,12 +1,42 @@
...
+[prow-staging.knative.dev](https://prow-staging.knative.dev) for testing the new
+config changes before rolling out to production, and all its config files are
+under [config/prow-staging](../config/prow-staging).
 
-1. Check under [prow/cluster](../config/prow/cluster) and add a new file there
-   if not exist.
+## Modify Prow cluster configs
+
+Prow is a collection of microservices that are deployed on a Kubernetes cluster,
+and all the Kubernetes object YAML files (except secrets) are under
```
Comment (chaodaiG) [inline guides/prow_setup.md:12]: Maybe `Kubernetes resources definitions`?

File: guides/prow_setup.md [hunk 4]
```diff
@@ -1,12 +1,42 @@
...
+[config/prow/cluster](../config/prow/cluster).
```
Comment (chaodaiG) [inline guides/prow_setup.md:13]: I think it's worthy mentioning that prod prow and staging prow configs are identical and this is controlled by auto-update process

File: guides/prow_knative_setup.md (hunk 2 above)
Comment (chizhg) [reply guides/prow_knative_setup.md:1]: Mostly, but also with a couple of changes and fixes.

File: guides/prow_setup.md (hunk 3 above)
Comment (chizhg) [reply guides/prow_setup.md:12]: I feel "Kubernetes object `.yaml` files" is more accurate. In https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/, I found  something like `In the .yaml file for the Kubernetes object you want to create`.

When you search for `Kubernetes resources definitions` , most of the results are about `Kubernetes Custom Resource Definitions`, and it seems normally we do not use this acronym for Kubernetes native resources.

Comment (chaodaiG) [reply guides/prow_setup.md:12]: Ah. I see, these are actually k8s services not resources

File: guides/prow_setup.md (hunk 4 above)
Comment (chizhg) [reply guides/prow_setup.md]: Done.

**Comment by chaodaiG** [comment]
/lgtm
Thanks!

**Pull request merged**